Sources/Purchasing/StoreKit2/ProductsFetcherSK2.swift

```diff
@@ -29,10 +29,17 @@ actor ProductsFetcherSK2 {
     func products(identifiers: Set<String>) async throws -> Set<SK2StoreProduct> {
         do {
             if let cachedProducts = await self.cachedProducts(withIdentifiers: identifiers) {
+                let cachedIdentifiers = Set(cachedProducts.map { $0.productIdentifier })
+                Logger.debug(Strings.offering.products_already_cached(identifiers: cachedIdentifiers))
                 return cachedProducts
             }
 
+            Logger.debug(
+                Strings.offering.no_cached_requests_and_products_starting_skproduct_request(identifiers: identifiers)
+            )
+
             let storeKitProducts = try await StoreKit.Product.products(for: identifiers)
+            Logger.rcSuccess(Strings.storeKit.skproductsrequest_received_response)
```

Comment:
Ditto here. Alternatively we could change these strings to be agnostic of the types? So something like "product request received response", etc.

Reply:
well we do have a completion handler up the call chain in `ProductsManager.sk2Products`:

` 
func sk2Products(withIdentifiers identifiers: Set<String>,
                     completion: @escaping (Result<Set<SK2StoreProduct>, Error>) -> Void) 
`

...so we could theoretically hold a map of identifiers -> handlers like `ProductsFetcherSK1` does 🤷